### PR TITLE
Internal: Add color control to element regression test

### PR DESCRIPTION
**Pull request by Nevoss**

### CODE REVIEW

File: tests/playwright/utils/controls/color.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+			}
+		}
+
+		await this.localToggleButton.click();
+
+		const value = await this.page.locator( '.pcr-app.visible input.pcr-result' ).inputValue();
+
+		await this.localToggleButton.click();
+
+		return [ this.constructor.LOCAL_COLOR_TYPE, value ];
```
Comment (StyleShit) [inline tests/playwright/utils/controls/color.js:51]: Maybe extract to `getGlobalColor` and `getLocalColor`?

File: tests/playwright/utils/controls/color.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
+const { ControlBase } = require( './control-base' );
+
+class Color extends ControlBase {
+	static GLOBAL_COLOR_TYPE = 'global';
+	static LOCAL_COLOR_TYPE = 'local';
```
Comment (StyleShit) [inline tests/playwright/utils/controls/color.js:5]:
```suggestion
	static TYPE_GLOBAL_COLOR = 'global';
	static TYPE_LOCAL_COLOR = 'local';
```

Or maybe:
```suggestion
	static COLOR_TYPE_GLOBAL = 'global';
	static COLOR_TYPE_LOCAL = 'local';
```

File: tests/playwright/utils/controls/color.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+				.click();
+
+			return;
+		}
+
+		await this.localToggleButton.click();
+
+		await this.page.locator( '.pcr-app.visible input.pcr-result' ).fill( value );
+
+		await this.localToggleButton.click();
```
Comment (StyleShit) [inline tests/playwright/utils/controls/color.js:69]: same as above. maybe extract to 2 methods

File: tests/playwright/utils/controls/color.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+			[ this.constructor.LOCAL_COLOR_TYPE, '#0000FF' ],
+		];
+
+		if ( await this.hasGlobalOption() ) {
+			const colorChooseDictonary = {
+				primary: 'secondary',
+				secondary: 'primary',
+				text: 'primary',
+				accent: 'primary',
+			};
```
Comment (StyleShit) [inline tests/playwright/utils/controls/color.js:86]: is it a "from -> to" type of thing?
if so, maybe we need a better name for that

**Review by StyleShit — CHANGES_REQUESTED** [review]

**Comment by StyleShit** [comment]
🔥

File: tests/playwright/utils/controls/color.js (hunk 1 above)
Comment (Nevoss) [reply tests/playwright/utils/controls/color.js:51]: Done

File: tests/playwright/utils/controls/color.js (hunk 2 above)
Comment (Nevoss) [reply tests/playwright/utils/controls/color.js:5]: Done

File: tests/playwright/utils/controls/color.js (hunk 3 above)
Comment (Nevoss) [reply tests/playwright/utils/controls/color.js:69]: Done

File: tests/playwright/utils/widgets/heading.js [hunk 5]
```diff
@@ -5,6 +5,12 @@ class Heading extends WidgetBase {
 		return 'heading';
 	}
 
+	getExcludeControls() {
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply tests/playwright/utils/widgets/heading.js:8]: Done

File: tests/playwright/utils/controls/color.js [hunk 6]
```diff
@@ -19,86 +19,111 @@ class Color extends ControlBase {
...
-			const colorChooseDictonary = {
-				primary: 'secondary',
-				secondary: 'primary',
-				text: 'primary',
-				accent: 'primary',
-			};
-
-			const value = originalType === this.constructor.GLOBAL_COLOR_TYPE
-				? colorChooseDictonary[ originalColor ]
+			const value = originalType === this.constructor.TYPE_GLOBAL_COLOR && 'primary' === originalColor
```
Comment (StyleShit) [inline tests/playwright/utils/controls/color.js:119]: maybe extract to conditional variable or at least add a comment?

**Review by StyleShit — CHANGES_REQUESTED** [review]

**Pull request merged**